app/controllers/news_items_controller.rb

```diff
@@ -1,9 +1,17 @@
 class NewsItemsController < ApplicationController
   def index
-    @news_items = NewsItem.published.to_a
+    @news_items = news_item_scope.to_a
   end
 
   def show
-    @news_item = NewsItem.published.find(params[:id])
+    @news_item = news_item_scope.where(slug: params[:slug]).first ||
+                 news_item_scope.where(id: params[:id]).first ||
+                 raise(ActiveRecord::RecordNotFound)
```

Comment:
You are right, just created a ticket for that #151
Until then, I am also going to leave the object loading methods as-is in the controller, I'll just add a comment to the method.